### PR TITLE
Use Arcade

### LINKED ISSUE #545: Release binaries have prerelease product version string
The Entity Framework nupkg (https://www.nuget.org/packages/EntityFramework/6.2.0) contains assemblies with a Windows product version (controlled by the assembly informational version) marking them as prerelease: `6.2.0-61023`

These assemblies are being flagged as prerelease in our release process. We are keeping `EntityFramework.dll` and `EntityFramework.SqlServer.dll` on a whitelist.

Would you please consider using `6.2.0+61023` or simply `6.2.0`? You often see release versions using build metadata similar to `6.2.0+build.61023.hash.abcdef01`.

### LINKED ISSUE #595: NuGet Package : Add `repository` metadata.
Basically, we can now add a repository metadata on a nuspec file to allow the repository to be surfaced on NuGet. This allows you to have a `projectUrl` as well a repository URL now.

I've talked about it [on Twitter](https://twitter.com/MaximRouiller/status/1030468873945075712) but the most basic things to know is here:

[NuSpec Reference](https://docs.microsoft.com/en-us/nuget/reference/nuspec?WT.mc_id=social-github-marouill#repository)

Reference Implementation in Json .NET
image

### PR DESCRIPTION
**Pull request by bricelam**
Changes:
- Remove SkipStrongNames
- Remove FxCop
- Remove StyleCop
- Test against LocalDB
- Use NuGet packages for
  - Microsoft.SqlServer.Types
  - PowerShell
  - VSSDK

Part of #612, resolves #595, resolves #545

Future work:
- Move Designer to another repo #694
- Port to xUnit.net 2
- Target .NET Core #271
- Enable NuGet PackageReference #189
- Enable PMC commands on .NET Core #231
- Enable Designer on .NET Core

### CODE REVIEW

**Comment by bricelam** [comment]
@Eilon Am I allowed to check in binaries from [SQL Server Compact 3.5 SP2](https://www.microsoft.com/en-us/download/details.aspx?id=5783) and the [Windows PowerShell 2.0 SDK](https://www.microsoft.com/en-us/download/details.aspx?id=2560)? They aren't available as NuGet packages...

**Comment by roji** [comment]
Uh, is there a document somewhere detailing the emoji specification for the team? It seems more complex and elaborate than I thought and I don't want to commit a fatal mistake...

**Comment by ErikEJ** [comment]
How about removing the SQL CE 3.5 provider ? - extended support ended 4/10/2018 https://support.microsoft.com/en-us/lifecycle/search/2855

**Comment by ajcvickers** [comment]
@roji Here's the specification:
* :shipit: Well-known GitHub ship-it emoji; used to be the normal way to approve before there was an `Approve` button.
* :sheep: :it: Literally "sheep IT" which we started using to mean, "I sign off on this but I haven't actually reviewed every file."
* :goat: :gb: Literally "goat GB". Me taking "sheep IT" to the next level because 3650 changes!

**Comment by roji** [comment]
That's.... awesome :)

**Comment by ajcvickers** [comment]
@ErikEJ We're considering not updating it. It's very low traffic anyway, the existing version should continue to work anyway, and the ADO.NET part is very unlikely to ever support .NET Core.

**Comment by bricelam** [comment]
Found [Microsoft.PowerShell.2.ReferenceAssemblies](https://www.nuget.org/packages/Microsoft.PowerShell.2.ReferenceAssemblies/).

**Comment by bricelam** [comment]
Hmm... Error on Ubuntu but not macOS:

> error MSB3245: Could not resolve this reference. Could not locate the assembly "System.Configuration".

**Comment by bricelam** [comment]
Ah, it changes case between net40 and net45 😅

**Comment by bricelam** [comment]
Submitted PR dsplaisted/ReferenceAssemblyPackages#2

**Comment by bricelam** [comment]
Looks like the agents don't have SQL Server Express.

**Comment by bricelam** [comment]
Keeping un-ported files around to make the diff more manageable. Note, anything under `eng/common/` is part of arcade and doesn't need to be reviewed.

**Comment by ErikEJ** [comment]
So bascially no product code changes?

**Comment by bricelam** [comment]
> SqlCeException : Unable to load the native components of SQL Server Compact corresponding to the ADO.NET provider of version 8876. Install the correct version of SQL Server Compact. Refer to KB article 974247 for more details.

Interesting. The NuGet package isn't enough?

**Comment by bricelam** [comment]
Oh yeah, we're consuming it via PackageReference. The files probably aren't being copied anymore.

**Comment by ErikEJ** [comment]
The old SQL CE package does not work with package reference, you need to copy the unmanaged DLLs "manually"

**Comment by bricelam** [comment]
3 tests failed (out of 1474):

**System.Data.Entity.Design.BasicDesignTimeScenarios.Uses_configuration_from_specified_assembly**
**System.Data.Entity.Design.BasicDesignTimeScenarios.Can_invoke_operations_accross_boundaries**
FileLoadException : Could not load file or assembly 'EntityFramework' or one of its dependencies. Strong name signature could not be verified.  The assembly may have been tampered with, or it was delay signed but not fully signed with the correct private key.

**FunctionalTests.Issue1776.FK_name_uniquification_is_not_triggered_on_different_tables**
SqlException : A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections.

**Comment by bricelam** [comment]
Looks like the functional tests take too long (>1hr) to use a public pool.

**Comment by ajcvickers** [comment]
@bricelam Can we provision our own pool?

**Comment by bricelam** [comment]
There is one provided by Arcade that we can use. By public, I mean provided by AzDO.

**Comment by bricelam** [comment]
@ajcvickers This is good-to-go if you want to skim through it one more time...

**Comment by ajcvickers** [comment]
@bricelam Any particular part you think I should look at more closely?

**Comment by bricelam** [comment]
Hmm... test code changes? Make sure I’m not skipping any critical coverage of something that should be passing

File: eng/common/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Don't touch this folder
+
+                uuuuuuuuuuuuuuuuuuuu
```
Comment (ajcvickers) [inline eng/common/README.md]: 🛑 Nice! 💯

**Comment by Eilon** [comment]
🏄

**Comment by Eilon** [comment]
🎮

**Pull request merged**